powerline_gitstatus/segments.py

```diff
@@ -77,38 +77,40 @@ def parse_status(self, lines):
 
         return (staged, unmerged, changed, untracked)
 
-    def build_segments(self, branch, detached, tag, behind, ahead, staged, unmerged, changed, untracked, stashed):
+    def build_segments(self, labels, branch, detached, tag, behind, ahead, staged, unmerged, changed, untracked, stashed):
         if detached:
             branch_group = 'gitstatus_branch_detached'
         elif staged or unmerged or changed or untracked:
             branch_group = 'gitstatus_branch_dirty'
         else:
             branch_group = 'gitstatus_branch_clean'
 
+        branch_label = u'' if detached else '%s ' % labels.get('branch', u'\ue0a0')
```

Comment:
When the working tree is in "detached" mode, we should not show the branch label since it's not on any branch.